include/abb_librws/common/rw/elog.h

```diff
@@ -0,0 +1,123 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <chrono>
+
+namespace abb ::rws ::rw ::elog
+{
+    // TODO: Replace with class that pulls domains?
+    /**
+     * @brief Elog domain enum
+     */
+    enum ElogDomain
+    {
+        COMMON = 0,
+        OPERATIONAL = 1,
+        SYSTEM = 2,
+        HARDWARE = 3,
+        PROGRAM = 4,
+        MOTION = 5,
+        IO_AND_COMMUNICATION = 7,
+        USER = 8,
+        SAFETY = 9,
+        INTERNAL = 10,
+        PROCESS = 11,
+        CONFIGURATION = 12,
+        RAPID_1 = 15,
+        RAPID_2 = 17
```

Comment:
Do these numbers match some constants defined somewhere, or are they just arbitrary? In the latter case I suggest just omitting numbers.

Also, we use `thisTypeOfCasing` for enum class members in e.g. `enum class ControllerState`, so let's be consistent.

Reply:
Yes, they are corresponding to the domains in [RobotWare](https://developercenter.robotstudio.com/api/rwsApi/elog_resources_page.html).
Those IDs might change in future or in some other RobotWare versions so they shouldn't be hardcoded in the end. I added [`getElogDomains`](https://github.com/NoMagicAi/abb_librws/blob/92e0e86898cf05b85e18dbf226c55ada605ddfc8/include/abb_librws/v2_0/rw/elog.h#L11) function that pulls IDs and names from controller using `/rw/elog` service.
I left those domains here as a placeholder and to be able to convert domain as number to name. I created a [task](https://nomagic.atlassian.net/browse/GRASP-13514) to fix it later. It is also the reason why I left the references to domains as `int` in other parts of code as those IDs might not be constant.